spharpy/samplings/coordinates.py

```diff
@@ -310,3 +314,29 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    def _set_weights(self, weights):
+        """Check and set sampling weights.
+
+        Set self._weights, which is an atleast_1d numpy array of shape
+        [L,M,...,N].
+
+        """
+        # check if weights is None
+        if weights is None:
+            self._weights = weights
+            return
+
```

Comment:
```suggestion
    if not isinstance(weights, (collections.abc.Sequence, np.ndarray)):
        raise TypeError("Weights must be array-like, not a %r" % type(weights))
```
If you pass an object array whose elements are all numeric, np will quietly convert them to floats.